crates/bitwarden-crypto/src/keys/master_key.rs

```
@@ -301,32 +320,30 @@ mod tests {
         let decrypted = master_key.decrypt_user_key(protected).unwrap();
 
         assert_eq!(
-            decrypted.key, user_key.0.key,
-            "Decrypted key doesn't match user key"
-        );
-        assert_eq!(
-            decrypted.mac_key, user_key.0.mac_key,
+            decrypted, user_key.0,
             "Decrypted key doesn't match user key"
         );
     }
 
     #[test]
     fn test_make_user_key2() {
-        let master_key = MasterKey(derive_symmetric_key("test1"));
+        let kdf_material = KdfDerivedKeymaterial {
+            key_material: if let SymmetricCryptoKey::Aes256CbcHmacKey(k) =
+                &derive_symmetric_key("test1")
+            {
+                k.encryption_key.clone()
+            } else {
+                panic!("Key is not an Aes256CbcHmacKey");
+            },
+        };
```

**suggestion:** Update `derive_symmetric_key` to return `Aes256CbcHmacKey`?

---

This doesn't seem to be updated?

---

Seems github ate my reply. I didn't update this because I don't think we want to change the signature of `derive_shareable_key` to always return a Aes256CbcHmacKey, because in the future in might return a XChaCha20Poly1305Key. 

So, I can change it here, but it might mean that we need to change it back in the future? Or pull in the logic in `derive_shareable_key` to the `derive_symmetric_key` to make sure the test keys always stay the same?

---

Actually, thinking about it this would need to be handled once those keys are introduced anyways, so let's care about it then. Fixed (39adb185)

---

Ok, leaving this issue up for whenever we actually add the new keys; Updated so it returns the `Aes256CbcHmacKey`. It does force wrapping with `SymmetricCryptoKey::Aes256CbcHmacKey()` in a few other places though.

---

I think that's actually a benefit if we do end up changing it to ChaCha? It forces us to ensure the usages are compatible vs it potentially silently failing?